### PR TITLE
[PP-4088] Add Python 3 support, update dev tooling, packaging

### PR DESCRIPTION
**Pull request by luislew**
This PR adds support for Python 3, and updates the dev tooling for an improved development experience.

To acknowledge the stability of APIs and long running usage by multiple projects, this will be the first `1.x` release for `versionalchemy`, with an accompanying `CHANGELOG.md` for tracking future changes.

---

Changes:
* Added support for Python 3.6+:
  * Use `six` to write Python 2/3 compatible code
  * Dropped `simplejson` dependency in favor of std lib `json` module
  * Updated to build a universal wheel
  * Added tests against Python 3.6 and 3.7 in Travis CI
* Improved dev tooling:
  * Updated local venv to use Python 3.7
  * Added `make deps` for regenerating `requirements.txt`
  * Added `make format` for auto-formatting code
  * Updated `make lint` to check Python 2/3 compatibility, import sorting
* Improved packaging:
  * Moved dev requirements into `extras_require[dev]`
  * Added classifiers, maintainer, `python_requires`
  * Moved version into its own `VERSION` file

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 - 2.7
+- 3.6
+- 3.7
 env:
 - SA_VERSION=1.0.19
-- SA_VERSION=1.1.13
-- SA_VERSION=1.2.0b2
+- SA_VERSION=1.1.18
+- SA_VERSION=1.2.19
+- SA_VERSION=1.3.8
```
Comment (luislew) [inline .travis.yml]: Updating to latest releases for each `1.x` SQLAlchemy branch, and added `1.3.x`

File: tests/test_utils.py [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 import sqlalchemy as sa
-from sqlalchemy.orm import sessionmaker
 from sqlalchemy.exc import StatementError
 from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
 
 from versionalchemy import utils
 
+EXCEPTIONS_NAME = 'builtins' if six.PY3 else 'exceptions'
+TYPE_NAME = 'class' if six.PY3 else 'type'
```
Comment (luislew) [inline tests/test_utils.py]: Python 2:
```
In [1]: ValueError.__module__
Out[1]: 'exceptions'

In [2]: 'value of type {} is not {}'.format(type({}), list)
Out[2]: "value of type <type 'dict'> is not <type 'list'>"
```

Python 3:
```
In [1]: ValueError.__module__
Out[1]: 'builtins'

In [2]: 'value of type {} is not {}'.format(type({}), list)                                         
Out[2]: "value of type <class 'dict'> is not <class 'list'>"
```

File: versionalchemy/utils.py [hunk 3]
```diff
@@ -181,13 +185,13 @@ class _JSONEncoded(TypeDecorator):
...
             return None
-        elif isinstance(value, basestring):
+        elif isinstance(value, six.string_types):
             value = json.loads(value)
 
         if self.json_type is not None and not isinstance(value, self.json_type):
             raise ValueError('value of type {} is not {}'.format(type(value), self.json_type))
 
-        return json.dumps(value, ensure_ascii=False, encoding='utf8', cls=VAJSONEncoder)
+        return json.dumps(value, ensure_ascii=False, cls=VAJSONEncoder)
```
Comment (luislew) [inline versionalchemy/utils.py]: `encoding='utf-8'` was already the default for `simplejson`, and it's _not_ a supported kwarg for `json.dumps`

File: versionalchemy/utils.py [hunk 4]
```diff
@@ -165,6 +167,8 @@ class VAJSONEncoder(json.JSONEncoder):
     def default(self, obj):
         if isinstance(obj, datetime.datetime):
             return obj.isoformat()
+        if isinstance(obj, decimal.Decimal):
```
Comment (ryankirkman) [inline versionalchemy/utils.py]: Out of interest, why was `Decimal` encoding added to `VAJSONEncoder`?

Comment (luislew) [reply versionalchemy/utils.py]: We have a test for encoding decimals as JSON; I’m not totally sure how it worked in Python 2.7

Comment (ryankirkman) [reply versionalchemy/utils.py]: Yeah, that’s confusing.

Either way, looks solid. The only thing I notice now is the docstring doesn’t quite match what the class does now. Specifically: https://github.com/NerdWalletOSS/versionalchemy/blob/db5f96e77d8dd35170ffd7a08fe0a6148183bdbc/versionalchemy/utils.py#L160-L163

Comment (luislew) [reply versionalchemy/utils.py]: For posterity, the reason this worked before is `simplejson.dumps` handles `Decimal` objects out of the box, while `json.dumps` does not.

**Pull request merged**